internal/search/query/parser.go

```diff
@@ -866,26 +868,65 @@ func (p *parser) ParsePattern(label labels) Pattern {
 // (-?)field:<string> where : matches the first encountered colon, and field
 // must match ^[a-zA-Z]+ and be allowed by allFields. Field may optionally
 // be preceded by '-' which means the parameter is negated.
-func (p *parser) ParseParameter() (Parameter, bool, error) {
+func (p *parser) ParseParameter(label labels) (Parameter, bool, error) {
 	start := p.pos
 	field, negated, advance := ScanField(p.buf[p.pos:])
 	if field == "" {
 		return Parameter{}, false, nil
 	}
 
 	p.pos += advance
-	value, labels, err := p.ParseFieldValue(field)
+	value, parsedLabels, err := p.ParseFieldValue(field)
 	if err != nil {
 		return Parameter{}, false, err
 	}
+
+	if label.IsSet(GlobFilters) {
+		switch field {
+		case "r", "repo", "f", "file":
+			value = globToRegex(value)
+		}
+	}
+
 	return Parameter{
 		Field:      field,
 		Value:      value,
 		Negated:    negated,
-		Annotation: Annotation{Range: newRange(start, p.pos), Labels: labels},
+		Annotation: Annotation{Range: newRange(start, p.pos), Labels: parsedLabels},
 	}, true, nil
 }
 
+func globToRegex(gob string) string {
```

Comment:
standardize on "glob"?

```suggestion
func globToRegex(glob string) string {
```